File: SAAMSONSDISTRIBUTORS/Models/Product.cs
﻿using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace SAAMSONSDISTRIBUTORS.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }

        [Column(TypeName = "nvarchar(8)")]
        public string? Code { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string? Name { get; set; }

        [Column(TypeName = "nvarchar(255)")]
        public string? Url { get; set; }
        
        [Column(TypeName = "decimal(18,2)")]
        public decimal? UnitPrice { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? SellingPrice { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string? SupplierName { get; set; }

        [Column(TypeName = "int")]
        public int? Quantity { get; set; }
    }
}

File: SAAMSONSDISTRIBUTORSWEB/sam-sons-distributors/src/app/models/product.ts
import { Decimal } from './common';

export interface Product {
  id: number;
  code?: string | null;
  name?: string | null;
  url?: string | null;
  unitPrice?: Decimal | null;     // decimal(18,2)
  sellingPrice?: Decimal | null;  // decimal(18,2)
  supplierName?: string | null;
  quantity?: number | null;       // stock on hand
}
